calico_cni_k8s_test.go

```diff
@@ -396,6 +396,91 @@ var _ = Describe("CalicoCni", func() {
 					}))
 				})
 			})
+
+			Context("Using host-local IPAM: request an IP then release it, and then request it again", func() {
+				It("Should successfully assign IP both times and successfully release it in the middle", func() {
+					netconfHostLocalIPAM := fmt.Sprintf(`
+				  {
+					"name": "net5",
+					  "type": "calico",
+					  "etcd_endpoints": "http://%s:2379",
+					  "ipam": {
+					    "type": "host-local",
+						"subnet": "usePodCidr"
+					  },
+				   "kubernetes": {
+				     "k8s_api_root": "http://127.0.0.1:8080"
+			    	},
+			   	  "policy": {"type": "k8s"},
+				  "log_level":"info"
+					}`, os.Getenv("ETCD_IP"))
+
+					config, err := clientcmd.DefaultClientConfig.ClientConfig()
+					Expect(err).NotTo(HaveOccurred())
+
+					clientset, err := kubernetes.NewForConfig(config)
+					Expect(err).NotTo(HaveOccurred())
+
+					// Create a K8s Node object with PodCIDR and name equal to hostname.
+					_, err = clientset.Nodes().Create(&v1.Node{
+						ObjectMeta: v1.ObjectMeta{Name: hostname},
+						Spec: v1.NodeSpec{
+							PodCIDR: "10.0.0.0/24",
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// Step 1: Create the first pod on node we created, and request a specific IP.
```

Comment:
That's what I was thinking of initially, but this comment doesn't really fit the `By()` BDD style, but I guess it makes sense to have it so it's clear when the tests fail